packages/legacy/src/queries/secrets.ts

```diff
@@ -13,11 +15,69 @@ export const useSecretQuery = (
   const secretResource = createSecretResource(namespace);
   return useMockableQuery<ISecret>(
     {
-      queryKey: ['secrets', secretName],
+      queryKey: ['secret', secretName],
       queryFn: async () => await consoleFetchJSON(secretResource.namedPath(secretName)),
       refetchInterval: usePollingContext().refetchInterval,
       enabled: !!secretName,
     },
-    secretName ==='secure' ? MOCK_SECRET_SECURE : MOCK_SECRET_INSECURE
+    secretName === 'secure' ? MOCK_SECRET_SECURE : MOCK_SECRET_INSECURE
   );
 };
+
+// function decodeSecretData (secret: ISecret) {
+//   if (secret.data) {
+//     for (const [key, value] of Object.entries(secret.data)) {
+//       secret.data[key] = atob(value);
+//     }
+//   }
+//   return secret;
+// }
+
+export const useSecretsQuery = (
+  secretNames: string[] | null,
```

Comment:
if `secretNames` can be really `null` then we need extra checks before each dot operator

Reply:
I'll refactor so the filtering won't happen if `secretNames` is `null` or empty.